presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -1049,4 +1084,33 @@ public StreamingSubPlan withBucketToPartition(Optional<int[]> bucketToPartition)
             return new StreamingSubPlan(fragment.withBucketToPartition(bucketToPartition), children);
         }
     }
+
+    private static class SqlStageExecutionInfo
+    {
+        private final SqlStageExecution stageExecution;
+        private final StageLinkage stageLinkage;
```

Comment:
Maybe add a comment stating `StageLinkage` links to the child stages? -- This is not related to this refactor but previously I need to think about whether it's up-linkage or down-linkage whenever I look into the code.

Reply:
hmmm... it links to both parent and children. it has the parent's ExchangeLocationsConsumer and the children's OutputBufferManagers.